src/components/BlockchainList/PostsList.tsx

```diff
@@ -50,15 +51,22 @@ export function PostListSuspended() {
       hasMore={amountOfLoadedPosts < postsAmount}
       loader={<LoadingText>Fetching more posts...</LoadingText>}
     >
-      {postsLoaded.map((post) => (
-        <Post
-          key={post.id}
-          blockchainId={Number(post.id)}
-          timestamp={Number(post.timestamp)}
-          text={post.post}
-          sender={post.sender}
-          derivativeAddress={post.derivativeAddress}
-        />
+      {postsLoaded.map((post, index) => (
+        <div className={scrollContainer}>
+          <Post
+            key={post.id}
+            blockchainId={Number(post.id)}
+            timestamp={Number(post.timestamp)}
+            text={post.post}
+            sender={post.sender}
+            derivativeAddress={post.derivativeAddress}
+          />
+          {postsAmount < 3 && postsAmount === index - 1 ? (
```

Comment:
This card is intended to be advertised after the third cast or at the end of casts (if there's less than 3 casts)
This is where the bug occurs - most likely because I added it as part of the infinite scroll. I'm wondering if I should list this as a post itself as part of postsLoaded..? Need help.